ocaml/xapi/xapi_host_helpers.ml

```diff
@@ -401,14 +401,15 @@ module Configuration = struct
       else
         hostname
     in
-    Printf.sprintf "InitiatorName=%s\nInitiatorAlias=%s\n" iqn hostname_chopped
+    [iqn; hostname_chopped]
 
   let set_initiator_name iqn =
     let hostname = Unix.gethostname () in
     let config_file = make_initiatorname_config iqn hostname in
-    Unixext.write_string_to_file
-      !Xapi_globs.iscsi_initiator_config_file
-      config_file
+    let shell_params = String.concat " " config_file in 
+    let shell_script = "sh /opt/xensource/libexec/set-iscsi-initiator" in 
+    let shell_command = String.concat " " [shell_script; shell_params] in
+    Sys.command shell_command
```

Comment:
We usually use forkhelpers to run commands.

Reply:
The name of the script should also be part of Xapi_globs, and part of the required scripts list.

Reply:
Thank you! I will update it.

Reply:
We have `Helpers.call_script`.